### PR TITLE
Add documentation for multiple stolon installations

### PR DESCRIPTION
**Pull request by UnderGreen**
----
How-to about building stolon application image and installing it in existing stolon cluster.
#

### CODE REVIEW

File: docs/multiple-stolon.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Installing multiple stolon clusters in one Gravity cluster
+This document provides an instruction on how to build and install multiple stolon clusters
```
Comment (a-palchikov) [inline docs/multiple-stolon.md:2]: provides instructions

File: docs/multiple-stolon.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Installing multiple stolon clusters in one Gravity cluster
```
Comment (a-palchikov) [inline docs/multiple-stolon.md:1]: ... stolon clusters into a Gravity cluster

File: docs/multiple-stolon.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+in one Gravity cluster.
+
+## Requirements
+
+- tele binary: 5.5+
+- stolon application: 1.13.0+
+
+## Building
+By default using provided [Makefile](../Makefile) stolon application is built as a Gravity application and could be installed in Gravity cluster with `gravity install` from resulting tarball.  
+For the additional stolon(s) in cluster [Application Catalog](https://gravitational.com/gravity/docs/catalog/) feature could be reused. Application does not need to be pushed to catalog, but the process of making of that application image type applies here.
```
Comment (a-palchikov) [inline docs/multiple-stolon.md:12]: `To add (or introduce) additional stolon instances into the cluster, use the [Application Catalog]() feature.`

Not sure I understood the following sentence though - do you mean to say that the same approach as implemented in the Makefile equally applies here?

File: docs/multiple-stolon.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+Stolon application contains [Helm](https://helm.sh/) chart in `resources/charts/stolon`. Custom values are stored in [custom-values.yaml](../resources/custom-values.yaml). They are important for installation in Gravity clusters and should be used as a starting point for your customization.
```
Comment (a-palchikov) [inline docs/multiple-stolon.md:14]: contains **a** [Helm]() chart ...
Custom values are important for the installation into a Gravity cluster. [custom-values.yaml]() contains a set of values that can be used as a starting point for further customizations.

File: docs/multiple-stolon.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+#### Building Docker images
+There is a special target in [Makefile](../Makefile) to build Docker images:
```
Comment (a-palchikov) [inline docs/multiple-stolon.md:17]: Which docker images? If there are multiple images I'd mention which one is for.

File: docs/multiple-stolon.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+```
+$ make images
+```
+
+Tag of the resulting images is automatically calculated from git by [version.sh](../version.sh) script and could be overriden by `VERSION` environment variable.
```
Comment (a-palchikov) [inline docs/multiple-stolon.md:23]: [version.sh]() automatically calculates a tag name to use for Docker image(s) based on a git tag. Use the VERSION environment variable to override the version from command line.

File: docs/multiple-stolon.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+```
+$ make images
+```
+
+Tag of the resulting images is automatically calculated from git by [version.sh](../version.sh) script and could be overriden by `VERSION` environment variable.  
+To build a Stolon application image with default custom-values file:
+
+#### Changing version of Helm chart
+Version of the stolon chart is hardcoded to `0.1.0` in [Chart.yaml](../resources/charts/stolon/Chart.yaml). You could replace it with command `sed -i "s/0.1.0/$(./version.sh)/g" resources/charts/stolon/Chart.yaml` or with any version you want to use.
```
Comment (a-palchikov) [inline docs/multiple-stolon.md:27]: Version of the stolon chart defaults to `0.1.0` but can be overridden by editing [Chart.yaml]().

File: docs/multiple-stolon.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+#### Building Gravity application image
+```
+$ tele build --values resources/custom-values.yaml \
+	--set registry="" --set tag=$(./version.sh) \
+	--version=$(./version.sh) resources/charts/stolon 
+```
+
+*Note*: If `VERSION` was overriden during the building of images then `--set tag=` should be set with the same value.
```
Comment (a-palchikov) [inline docs/multiple-stolon.md:36]: If you change the version (e.g. via VERSION environment variable), set the tag to the same value with `--set tag=<version>`

File: docs/multiple-stolon.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+#### Building Gravity application image
+```
+$ tele build --values resources/custom-values.yaml \
+	--set registry="" --set tag=$(./version.sh) \
+	--version=$(./version.sh) resources/charts/stolon 
+```
+
+*Note*: If `VERSION` was overriden during the building of images then `--set tag=` should be set with the same value.  
+Command produces tarball with vendored images and Helm chart inside. Default name of the tarball will be stolon-app-$(VERSION).tar in the root directory of the repository. Flag `-o(--output) filename.tar` could be added to `tele build` command to specify the name of a resulting tarball.
```
Comment (a-palchikov) [inline docs/multiple-stolon.md:37]: As specified, the above command will create a tarball named stolon-app-$(VERSION).tar in the repository's root directory with Docker images and Helm charts bundled inside. To change the name of the output file, use `--output=<new-name>`.

File: docs/multiple-stolon.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+## Installation
+Resulted tarball with the application should be uploaded to Gravity cluster along with the [custom-values.yaml](../resources/custom-values.yaml) files because the last one needs to be used for setting custom Chart values.
+
+``` shell
+# gravity app install --values custom-values.yaml --set clusterName=kube-stolon2 --set keeperDataPath=/var/lib/data/stolon2 --name stolon2 stolon-app.tar
+```
+There are two important values that must be set for the second(or more) stolon in Gravity cluster: `clusterName` and `keeperDataPath`.
```
Comment (a-palchikov) [inline docs/multiple-stolon.md:45]: Each subsequent instance of stolon requires setting the values of `clusterName` and `keeperDataPath`.

File: docs/multiple-stolon.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+- `clusterName` specifies the path in Etcd to store data about the stolon cluster. Name should be unique for each stolon cluster.
```
Comment (a-palchikov) [inline docs/multiple-stolon.md:46]: `clusterName` specifies a path in etcd that the stolon instance will use for its data. The path must be unique for each stolon cluster.
`keeperDataPath` specifies a path on host each instance will use to store data. The path should be unique for each stolon cluster.

Let me know if I got the meaning correct/wrong.

File: docs/multiple-stolon.md (hunk 3 above)
Comment (UnderGreen) [reply docs/multiple-stolon.md:12]: I mean our Application catalog instructions contain pushing an app to a Hub, but, technically, it is not needed. We can just create tarball and use it with `gravity app install app-version.tar`.

**Review by a-palchikov — APPROVED** [review]

**Pull request merged**